eth/api_test.go

```diff
@@ -28,6 +30,31 @@ import (
 
 var dumper = spew.ConfigState{Indent: "    "}
 
+func TestAccountRangeAt(t *testing.T) {
+	var (
+		state, _ = state.New(common.Hash{}, state.NewDatabase(ethdb.NewMemDatabase()))
+		addrs = [512]common.Address{}
+	)
+
+	for i := 0; i < 512; i++ {
+		addr := fmt.Sprintf("%x", i)
+		addrs[i] = common.HexToAddress(addr)
+	}
+
+	for i := range addrs {
+		state.SetBalance(addrs[i], big.NewInt(1))
+	}
+
+	// test retrieving less than max results
+
+/*
+	result := accountRangeAt(state.Trie(), common.Address{0x0}, 128)
```

Comment:
@gballet Trying to think of how to get the state trie here...  I'm not sure adding an accessor would be appropriate if it's only going to be used for this test case.

Reply:
you know what... I swear I've actually been able to do this previously without making the trie field public.  Let me look back and see what I did...

Reply:
@jwasinger did you find it or do you still need some input ?

Reply:
I don't have the code anymore and I'm trying to figure out exactly how to construct the Trie.. not blocked but it will take me a bit to figure out which pieces I need for it to work.

Reply:
Okay seems to be working.. kinda.  I get weird output for the addresses:

```
    api_test.go:69: 0x0000000000000000000000000000000000000000                                                                                                                                                       
    api_test.go:69: 0x00000000000000000000000000000000000000A5                                                                                                                                                       
    api_test.go:69: 0x00000000000000000000000000000000000001c3                                                                                                                                                       
    api_test.go:69: 0x00000000000000000000000000000000000000F0                                                                                                                                                       
    api_test.go:69: 0x000000000000000000000000000000000000001F                                                                                                                                                       
    api_test.go:69: 0x00000000000000000000000000000000000001c5                                                                                                                                                       
    api_test.go:69: 0x0000000000000000000000000000000000000054                                                                                                                                                       
    api_test.go:69: 0x000000000000000000000000000000000000017f                                                                                                                                                       
    api_test.go:69: 0x00000000000000000000000000000000000000B1                                                                                                                                                       
    api_test.go:69: 0x000000000000000000000000000000000000002c                                                                                                                                                       
    api_test.go:69: 0x0000000000000000000000000000000000000003                                                                                                                                                       
    api_test.go:69: 0x0000000000000000000000000000000000000029                                                                                                                                                       
    api_test.go:69: 0x000000000000000000000000000000000000015e                                                                                                                                                       
    api_test.go:69: 0x00000000000000000000000000000000000001fa                                                                                                                                                       
    api_test.go:69: 0x00000000000000000000000000000000000000f3                                                                                                                                                       
    api_test.go:69: 0x00000000000000000000000000000000000001B6                                                                                                                                                       
    api_test.go:69: 0x0000000000000000000000000000000000000164                                                                                                                                                       
    api_test.go:69: 0x0000000000000000000000000000000000000151                                                                                                                                                       
    api_test.go:69: 0x000000000000000000000000000000000000009D                                                                                                                                                       
    api_test.go:69: 0x00000000000000000000000000000000000001A3                                                                                                                                                       
    api_test.go:69: 0x00000000000000000000000000000000000000D4                                                                                                                                                       
    api_test.go:69: 0x000000000000000000000000000000000000012c                                                                                                                                                       
    api_test.go:69: 0x00000000000000000000000000000000000001E8                                                                                                                                                       
    api_test.go:69: 0x00000000000000000000000000000000000000C9                                                                                                                                                       
    api_test.go:69: 0x0000000000000000000000000000000000000027                                                                                                                                                       
    api_test.go:69: 0x0000000000000000000000000000000000000108                                                                                                                                                       
    api_test.go:69: 0x00000000000000000000000000000000000001B1                                                                                                                                                       
    api_test.go:69: 0x00000000000000000000000000000000000000AC                                                                                                                                                       
    api_test.go:69: 0x0000000000000000000000000000000000000034                                                                                                                                                       
    api_test.go:69: 0x00000000000000000000000000000000000001C4                                                                                                                                                       
    api_test.go:69: 0x00000000000000000000000000000000000000A2                                                                                                                                                       
    api_test.go:69: 0x000000000000000000000000000000000000004A                                                                                                                                                       
    api_test.go:69: 0x00000000000000000000000000000000000000fb                                                                                                                                                       
    api_test.go:69: 0x00000000000000000000000000000000000000D5                                                                                                                                                       
    api_test.go:69: 0x00000000000000000000000000000000000001bB
    ...                                                                                                                                               
```

Reply:
New unit tests failing :/ Somehow the trie doesn't contain as many accounts as it should.

Reply:
There are no duplicate accounts in the test vector.

Reply:
I think it might have to do with the zero padding in front of these addresses (and some subtlety of the trie that I am forgetting).  I'll try with randomly generated addresses tomorrow.

Reply:
Randomly generated addresses resulting in the same issue.... Have to look into this.  @gballet if you have any insights here, they would be helpful.